app/models/herbarium.rb

```diff
@@ -58,147 +64,16 @@ class Herbarium < AbstractModel
-    "GB" => nil, # University of Gothenburg
-    "HAW-F" => nil, # University of Hawaii, Joseph F. Rock Herbarium
-    "ILL" => nil, # University of Illinois Herbarium
-    "ILLS" => nil, # University of Illinois, Illinois Natural History Survey Fungarium
-    "KANU-KU-F" => nil, # University of Kansas, R. L. McGregor Herbarium
-    "MAINE" => nil, # University of Maine, Richard Homola Mycological Herbarium
-    "WIN" => nil, # University of Manitoba
-    "MICH" => nil, # University of Michigan Herbarium
-    "MIN" => nil, # University of Minnesota, Bell Museum of Natural History Herbarium Fungal Collection
-    "MISS" => nil, # University of Mississippi
-    "MONTU" => nil, # University of Montana Herbarium
-    "NEB" => nil, # University of Nebraska State Museum, C.E. Bessey Herbarium - Fungi
-    "UNM-Fungi" => nil, # University of New Mexico Herbarium Mycological Collection
-    "UNCA-UNCA" => nil, # University of North Carolina Asheville
-    "NCU-Fungi" => nil, # University of North Carolina at Chapel Hill Herbarium: Fungi
-    "O" => nil, # University of Oslo, Natural History Museum Fungarium
-    "URV" => nil, # University of Richmond
-    "USAM" => nil, # University of South Alabama Herbarium
-    "USCH-Fungi" => nil, # University of South Carolina, A. C. Moore Herbarium Fungal Collection
-    "USF" => nil, # University of South Florida Herbarium - Fungi including lichens
-    "TU" => nil, # University of Tartu Natural History Museum
-    "TENN-F" => nil, # University of Tennessee Fungal Herbarium
-    "UCHT-F" => nil, # University of Tennessee, Chattanooga
-    "TEX" => nil, # University of Texas Herbarium
-    "VT" => nil, # University of Vermont, Pringle Herbarium, Macrofungi
-    "WTU" => nil, # University of Washington Herbarium
-    "UWAL" => nil, # University of West Alabama Fungarium
-    "WIS" => nil, # University of Wisconsin-Madison Herbarium
-    "UWSP" => nil, # University of Wisconsin-Stevens Point Herbarium
-    "RMS" => nil, # University of Wyoming, Wilhelm G. Solheim Mycological Herbarium
-    "UPS-BOT" => nil, # Uppsala University, Museum of Evolution
-    "USAC-USCG Hongos" => nil, # Usac, Cecon, Herbario USCG Hongos
-    "CFMR" => nil, # USDA Forest Service, Center for Forest Mycology Research
-    "FPF" => nil, # USDA Forest Service, Rocky Mountain Research Station
-    "BPI" => nil, # USDA United States National Fungus Collections
-    "VSC" => nil, # Valdosta State University Herbarium
-    "VPI" => nil, # Virginia Tech University, Massey Herbarium - Fungi
-    "YSU-F" => nil # Yugra State University Fungarium, specimen-based
-  }.freeze
-  # rubocop:enable Layout/LineLength
+  def self.mcp_collections
+    @mcp_collections ||=
+      begin
+        csv_path = Rails.public_path.join("mycoportal-collections.csv")
+        CSV.foreach(csv_path,
+                    headers: true).each_with_object([]) do |row, collections|
+          collections << row.to_h
+        end
+      end
+  end
```

Comment:
@mo-nathan, @nimmolo:
1. Does this method make sense? (I'm still uncomfortable with some aspects of Ruby.)
2. Is there a more performant way to deal with `mcp_collections`?